include/RAJA/exec-cuda/scan_cuda.hxx

```diff
@@ -103,6 +103,16 @@ struct minimum : public ::thrust::minimum<T> {
 
 }  // closing brace for cuda_scan namespace
 
+namespace scan
+{
+
+template <typename T>
+struct defaults <cuda_exec, T> {
```

Comment:
@DavidPoliakoff I created `RAJA::operators`

- implemented most of the binary associative operators we'd use (minimum, maximum, plus, multiplies, logical_and, logical_or) along with others non-associative or unary operators
- marked operator() as `RAJA_HOST_DEVICE`
- associative operators extend from `detail::associative_tag` and have a `static constexpr const T identity` value defining the identity value for the operator